File: src/main/java/com/epicsquid/whoosh/init/WhooshItems.java
package com.epicsquid.whoosh.init;

import com.epicsquid.whoosh.Whoosh;
import com.epicsquid.whoosh.item.TransporterItem;
import com.tterrag.registrate.Registrate;
import com.tterrag.registrate.util.entry.ItemEntry;

public class WhooshItems {

	private static final Registrate REGISTRATE = Whoosh.registrate();

	public static final ItemEntry<TransporterItem> TRANSPORTER = REGISTRATE.item("transporter", props -> new TransporterItem(props, 10000, 1000))
			.tab(() -> Whoosh.CREATIVE_TAB)
			.register();

	public static void classload() {
	}
}

File: src/main/java/com/epicsquid/whoosh/registery/WhooshItems.kt
package com.epicsquid.whoosh.registery

import com.epicsquid.whoosh.item.TransporterItem
import com.tterrag.registrate.Registrate
import com.tterrag.registrate.util.entry.ItemEntry
import com.tterrag.registrate.util.nullness.NonNullFunction
import com.tterrag.registrate.util.nullness.NonNullSupplier
import dev.epicsquid.whoosh.Whoosh
import dev.epicsquid.whoosh.Whoosh.Companion.MODID
import net.minecraft.world.item.CreativeModeTab
import net.minecraft.world.item.Item

object WhooshItems {
    private val REGISTRATE: Registrate = Whoosh.registrate
    val TRANSPORTER: ItemEntry<TransporterItem> = REGISTRATE.item<TransporterItem>("transporter",
        NonNullFunction<Item.Properties, TransporterItem> { props: Item.Properties? ->
            TransporterItem(
                props, 10000, 1000
            )
        })
        //add item to Creative mode tab

        .tab(NonNullSupplier<CreativeModeTab> { CreativeModeTab(MODID)})
        .register()

    fun init() {}
}
